url9d1719148e_mrqsdavi_pegmachine_tgz-pJ8-peg_RegexJ8/src/peg/AnalizadorLexico.java

```diff
@@ -0,0 +1,701 @@
+package peg;
+/* The following code was generated by JFlex 1.4.3 on 19/11/13 14:30 */
+
+import java.io.IOException;
+import java_cup.runtime.*; 
+
+
+/**
+ * This class is a scanner generated by 
+ * <a href="http://www.jflex.de/">JFlex</a> 1.4.3
+ * on 19/11/13 14:30 from the specification file
+ * <tt>/Users/Davi/Documents/PEGMachine/PEGMachine/src/Expressoes.flex</tt>
+ */
+class AnalizadorLexico implements java_cup.runtime.Scanner {
+
+  /** This character denotes the end of file */
+  public static final int YYEOF = -1;
+
+  /** initial size of the lookahead buffer */
+  private static final int ZZ_BUFFERSIZE = 16384;
+
+  /** lexical states */
+  public static final int YYINITIAL = 0;
+
+  /**
+   * ZZ_LEXSTATE[l] is the state in the DFA for the lexical state l
+   * ZZ_LEXSTATE[l+1] is the state in the DFA for the lexical state l
+   *                  at the beginning of a line
+   * l is of the form l = 2*k, k a non negative integer
+   */
+  private static final int ZZ_LEXSTATE[] = { 
+     0, 0
+  };
+
+  /** 
+   * Translates characters to character classes
+   */
+  private static final String ZZ_CMAP_PACKED = 
+    "\11\0\1\22\1\23\2\0\1\22\22\0\1\22\1\14\1\0\3\0"+
+    "\1\15\1\11\1\1\1\2\1\7\1\10\1\0\1\20\1\13\1\12"+
+    "\1\25\11\25\2\0\1\17\2\0\1\16\1\0\6\24\24\24\1\3"+
+    "\1\0\1\4\1\0\1\24\1\0\6\24\2\24\1\24\4\24\1\24"+
+    "\1\24\1\24\1\24\1\24\1\24\1\24\3\24\1\24\2\24\1\5"+
+    "\1\0\1\6\1\21\uff81\0";
+
+  /** 
+   * Translates characters to character classes
+   */
+  private static final char [] ZZ_CMAP = zzUnpackCMap(ZZ_CMAP_PACKED);
+
+  /** 
+   * Translates DFA states to action switch labels.
+   */
+  private static final int [] ZZ_ACTION = zzUnpackAction();
+
+  private static final String ZZ_ACTION_PACKED_0 =
+    "\1\1\1\2\1\3\1\4\1\2\1\5\1\6\1\7"+
+    "\1\10\1\2\1\11\1\12\1\13\1\14\1\15\1\2"+
+    "\1\16\1\1\1\17\1\20\1\0\1\21\1\0\1\22"+
+    "\1\23";
+
+  private static int [] zzUnpackAction() {
+    int [] result = new int[25];
+    int offset = 0;
+    offset = zzUnpackAction(ZZ_ACTION_PACKED_0, offset, result);
+    return result;
+  }
+
+  private static int zzUnpackAction(String packed, int offset, int [] result) {
+    int i = 0;       /* index in packed string  */
+    int j = offset;  /* index in unpacked array */
+    int l = packed.length();
+    while (i < l) {
+      int count = packed.charAt(i++);
+      int value = packed.charAt(i++);
+      do result[j++] = value; while (--count > 0);
+    }
+    return j;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 70:78, 131:140, 173:181.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.